clients/drcachesim/tests/view_test.cpp

```diff
@@ -586,11 +586,63 @@ run_chunk_tests(void *drcontext)
     return run_single_thread_chunk_test(drcontext) && run_serial_chunk_test(drcontext);
 }
 
+/* Test view tool on a OFFLINE_FILE_TYPE_ARCH_REGDEPS trace.
+ * The trace hardcoded in entries is X64, so we only test on X64 architectures here.
+ */
+bool
+run_regdeps_test(void *drcontext)
+{
+#ifdef X64
+    const memref_tid_t t1 = 3;
+    std::vector<memref_tid_t> tids = { t1 };
+    constexpr addr_t PC = 0x00007f6fdd3ec360;
+    constexpr addr_t ENCODING_REGDEPS_ISA = 0x0006090600010011;
+    std::vector<std::vector<trace_entry_t>> entries = { {
+        { TRACE_TYPE_HEADER, 0, { 0x1 } },
+        { TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_VERSION, { 3 } },
+        { TRACE_TYPE_MARKER,
+          TRACE_MARKER_TYPE_FILETYPE,
+          { OFFLINE_FILE_TYPE_ARCH_REGDEPS | OFFLINE_FILE_TYPE_ENCODINGS |
+            OFFLINE_FILE_TYPE_SYSCALL_NUMBERS | OFFLINE_FILE_TYPE_BLOCKING_SYSCALLS } },
+        { TRACE_TYPE_THREAD, 0, { t1 } },
+        { TRACE_TYPE_PID, 0, { t1 } },
+        { TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_CACHE_LINE_SIZE, { 64 } },
+        { TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_CHUNK_INSTR_COUNT, { 2 } },
+        { TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_TIMESTAMP, { 1002 } },
+        { TRACE_TYPE_MARKER, TRACE_MARKER_TYPE_CPU_ID, { 2 } },
+        { TRACE_TYPE_ENCODING, 8, { ENCODING_REGDEPS_ISA } },
+        { TRACE_TYPE_INSTR, 3, { PC } },
+        { TRACE_TYPE_FOOTER, 0, { 0 } },
+    } };
+    std::string expect =
+        std::string(R"DELIM(           1           0:           3 <marker: version 3>
+           2           0:           3 <marker: filetype 0x20e00>
+           3           0:           3 <marker: cache line size 64>
+           4           0:           3 <marker: chunk instruction count 2>
+           5           0:           3 <marker: timestamp 1002>
+           6           0:           3 <marker: tid 3 on core 2>
+           7           1:           3 ifetch       )DELIM") +
+        std::string(R"DELIM(3 byte(s) @ 0x00007f6fdd3ec360 )DELIM") +
+        std::string(
+            R"DELIM(11 00 01 00 06 09 06 00 move   %rv4[8byte] -> %rv7[8byte]
```

Comment:
Probably there should be core IR tests independent of this drmemtrace view tool (could be a separate PR for the core IR, split from the drmemtrace view tool) in suite/test/api/.

Reply:
Ok, will add tests in suite/test/api/dis* in a separate PR.

Reply:
What I meant was one PR for core disassembly + tests, and a separate PR for the logically separate drmemtrace view tool then using the new disasm.  Changes to the core should have their own tests.  The drmemtrace tool is a logically separate piece.

Reply:
Oh, sorry about that, I missed the keyword: "split" in your first comment.
I'll split this PR in "disassembly + tests in suite/tests/api" and "view tool + current view_test.cpp on a regdeps trace".

Reply:
Done. This PR is only for core disassembly of regdeps instructions + tests.
The PR for the view tool + test is ready, once this one goes into master.